File: Sources/vibetrail/Format.swift
import Foundation
import VibeTrailCore

enum Format {
    static func relativeTime(_ date: Date) -> String {
        let interval = Date().timeIntervalSince(date)
        switch interval {
        case ..<60: return "just now"
        case ..<3600: return "\(Int(interval / 60))m ago"
        case ..<86400: return "\(Int(interval / 3600))h ago"
        case ..<(86400 * 30): return "\(Int(interval / 86400))d ago"
        default:
            let formatter = DateFormatter()
            formatter.dateFormat = "yyyy-MM-dd"
            return formatter.string(from: date)
        }
    }

    static func duration(_ interval: TimeInterval) -> String {
        switch interval {
        case ..<1: return "-"
        case ..<60: return "\(Int(interval))s"
        case ..<3600: return "\(Int(interval / 60))m"
        default: return String(format: "%.1fh", interval / 3600)
        }
    }

    static func abbreviatePath(_ path: String) -> String {
        let home = FileManager.default.homeDirectoryForCurrentUser.path
        return path.hasPrefix(home) ? "~" + path.dropFirst(home.count) : path
    }

    static func truncate(_ text: String, to length: Int) -> String {
        text.count > length ? String(text.prefix(length - 1)) + "…" : text
    }

    static func roleIcon(_ role: Role) -> String {
        switch role {
        case .user: return "❯"
        case .assistant: return "●"
        case .system: return "◦"
        }
    }

    /// Fixed-width left-aligned column padding (rough; CJK width ignored).
    static func pad(_ text: String, _ width: Int) -> String {
        text.count >= width ? text : text + String(repeating: " ", count: width - text.count)
    }
}

File: crates/vibetrail-cli/src/format.rs
use chrono::{DateTime, Utc};
use vibetrail_core::Role;

pub fn relative_time(date: &DateTime<Utc>) -> String {
    let seconds = (Utc::now() - *date).num_seconds();
    match seconds {
        s if s < 60 => "just now".to_string(),
        s if s < 3600 => format!("{}m ago", s / 60),
        s if s < 86_400 => format!("{}h ago", s / 3600),
        s if s < 86_400 * 30 => format!("{}d ago", s / 86_400),
        _ => date.format("%Y-%m-%d").to_string(),
    }
}

pub fn duration(seconds: f64) -> String {
    match seconds {
        s if s < 1.0 => "-".to_string(),
        s if s < 60.0 => format!("{}s", s as u64),
        s if s < 3600.0 => format!("{}m", (s / 60.0) as u64),
        s => format!("{:.1}h", s / 3600.0),
    }
}

pub fn abbreviate_path(path: &str) -> String {
    match dirs::home_dir() {
        Some(home) => {
            let home = home.to_string_lossy();
            match path.strip_prefix(home.as_ref()) {
                Some(rest) => format!("~{rest}"),
                None => path.to_string(),
            }
        }
        None => path.to_string(),
    }
}

pub fn truncate(text: &str, length: usize) -> String {
    if text.chars().count() > length {
        let mut truncated: String = text.chars().take(length - 1).collect();
        truncated.push('…');
        truncated
    } else {
        text.to_string()
    }
}

pub fn role_icon(role: Role) -> &'static str {
    match role {
        Role::User => "❯",
        Role::Assistant => "●",
        Role::System => "◦",
    }
}

/// Fixed-width left-aligned column padding (rough; CJK width ignored).
pub fn pad(text: &str, width: usize) -> String {
    let count = text.chars().count();
    if count >= width {
        text.to_string()
    } else {
        format!("{text}{}", " ".repeat(width - count))
    }
}
